src/query_ctrl.js

```diff
@@ -9,13 +9,22 @@ export class GenericDatasourceQueryCtrl extends QueryCtrl {
     this.scope = $scope;
     this.target.target = this.target.target || 'select metric';
     this.target.type = this.target.type || 'timeseries';
-    this.target.data = this.target.data || null;
+    this.target.data = this.target.data || "";
+
+    // special handling when in table panel
+    if (!this.target.format) {
+      this.target.format = this.panelCtrl.panel.type === 'table' ? 'table' : 'timeseries';
```

Comment:
I have changed it to `time_series` according to https://github.com/grafana/simple-json-datasource/pull/103#issuecomment-412503403

Reply:
If ok I've prefer timeseries for readability. With the mysql source its only internal while with JSON its actually being sent to the data source.

Reply:
So shall we use `timeseries` then?

Reply:
yes :)